### PR TITLE
[NavigationDrawer] Allow clients to expand/collapse drawer

### PR DESCRIPTION
**Pull request by codeman7**
This will allow clients to expand or collapse the drawer to the full height of their `headerViewController.preferredContentSize.height` + `contentViewController.preferredContentSize.height`. If the height of the two combined is 300 on a 400pt screen this this expands the drawer to 75%. If the height is 1000 on a 400pt screen this expands the drawer to 100%. 

This is a prototype, it is missing test and docs. 

This is an alternative to #5826

### CODE REVIEW

**Comment by yarneo** [comment]
is this still a prototype and missing docs as the description states?

also, can you please provide a gif showing it work correctly for small content being moved to top

File: components/NavigationDrawer/examples/supplemental/BottomDrawerSupplemental.swift [hunk 1]
```diff
@@ -17,7 +17,7 @@ import MaterialComponents.MaterialColorScheme
 import MaterialComponents.MaterialNavigationDrawer
 
 class DrawerContentViewController: UIViewController {
-  var preferredHeight: CGFloat = 2000
```
Comment (yarneo) [inline components/NavigationDrawer/examples/supplemental/BottomDrawerSupplemental.swift:20]: please revert this change

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h [hunk 2]
```diff
@@ -101,4 +101,18 @@
  */
 - (void)setContentOffsetY:(CGFloat)contentOffsetY animated:(BOOL)animated;
 
+/**
+ Animates the navigation drawer to the full height of it's content, if the content is larger than
```
Comment (yarneo) [inline components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h:105]: is this true? it is supposed to animate the drawer to full screen no matter the content height

**Comment by romoore** [comment]
If this is a prototype, @jverkoey  has said it doesn't require full code review. It only has to build and pass CI as far as I know.
